### ISSUE TITLE
Implement `on_attester_slashing` handler

### ISSUE THREAD

**Issue by MegaRedHand** [lambdaclass/lambda_ethereum_consensus#378]
According to the spec:

```python
def on_attester_slashing(store: Store, attester_slashing: AttesterSlashing) -> None:
    """
    Run ``on_attester_slashing`` immediately upon receiving a new ``AttesterSlashing``
    from either within a block or directly on the wire.
    """
    attestation_1 = attester_slashing.attestation_1
    attestation_2 = attester_slashing.attestation_2
    assert is_slashable_attestation_data(attestation_1.data, attestation_2.data)
    state = store.block_states[store.justified_checkpoint.root]
    assert is_valid_indexed_attestation(state, attestation_1)
    assert is_valid_indexed_attestation(state, attestation_2)

    indices = set(attestation_1.attesting_indices).intersection(attestation_2.attesting_indices)
    for index in indices:
        store.equivocating_indices.add(index)
```

Some features might be missing, but the idea is to start integrating what we have.

**Comment by ayushm2003** [comment]
can I take this up?

**Comment by MegaRedHand** [comment]
Sorry, I didn't notice your messages! I already implemented the handler in #404
